### ISSUE TITLE
Faulty calculation in UncommunicativeName

### ISSUE THREAD

**Issue by marcandre** [rubocop-hq/rubocop#8229]
I didn't check this cop closely, but just looking at the code, there are multiple errors in the calculation of the offending range.

I think the following examples will lead to the wrong selection range: `any_offensive_name_with_underscore`, `*anyoffensivename`, `**anyoffensivename`.

Code can probably simply rely on `location.name` (not sure, didn't check all dependents)

**Comment by harrylewis** [comment]
Hey @marcandre 👋  I'm interested in picking this up.

I just want to confirm the buggy behaviour first. I created a file that looks like the following.

```ruby
# frozen_string_literal: true

def test(_x); end
```

Running `rubocop` on this file produces the following offense.

```
Inspecting 1 file
C

Offenses:

app.rb:5:10: C: Naming/MethodParameterName: Method parameter must be at least 3 characters long.
def test(_x); end
         ^

1 file inspected, 1 offense detected
```

The expected behaviour should be that the calculated location is over all of `_x`, right?

```
Inspecting 1 file
C

Offenses:

app.rb:5:10: C: Naming/MethodParameterName: Method parameter must be at least 3 characters long.
def test(_x); end
         ^^

1 file inspected, 1 offense detected
```

I see similar issues when using `*` and `**`.
